rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -119,7 +119,7 @@ object Reduce {
         )
       )
 
-  private def spatialMatchAndCharge[M[_]: Sync](target: Par, pattern: Par)(
+  def spatialMatchAndCharge[M[_]: Sync](target: Par, pattern: Par)(
```

Comment:
here we should extract the cost-accounting stuff in a function and test that separately. I'm thinking of lines 128-142

Reply:
This function was created to encapsulate the cost of the spatial match and charging for it.